### ISSUE TITLE
Render reedbed with a non-transparent color

### ISSUE THREAD

**Issue by Klaus-Tockloth** [gravitystorm/openstreetmap-carto#2013]
An area of reedbed is currently rendered as an overlay with a transparent color. This is imho incorrect because one doesn't see any water.

Example (the brown area in the west and north is reedbed):
![bildschirmfoto 2016-01-02 um 08 45 23](https://cloud.githubusercontent.com/assets/5493128/12073739/7c3af730-b12e-11e5-9e03-92678129d9a1.jpg)

The current rendering result (https://www.openstreetmap.org/#map=14/54.8719/8.3219):

![bildschirmfoto 2016-01-02 um 08 46 34](https://cloud.githubusercontent.com/assets/5493128/12073742/9aad8b06-b12e-11e5-8634-4fceb1886224.jpg)

I recommend to render reedbed this way (example):

![bildschirmfoto 2016-01-02 um 08 49 24](https://cloud.githubusercontent.com/assets/5493128/12073744/a5b3fcec-b12e-11e5-965f-7a7e954c042b.jpg)

Regards Klaus

**Comment by imagico** [comment]
Currently this style is trying to use pattern only rendering without a fill color for wetlands where the ground is sometimes/partly water covered.  This is the case for reed which grows to considerable water depth - see:

https://en.wikipedia.org/wiki/Phragmites

The problem here is more that mapping tends to put the waterline at the outer edge of the reed bed and the actual water line within the reed is not mapped.  This is perfectly understandable and also makes sense from a certain perspective but strictly seen there is no convention in OSM mandating that.

More generally color fills are also problematic for wetlands at the moment since they render differently when overlapping with ocean compared to overlapping with natural=water.  This is a problem for tidal flats for example.  Once the move to water polygons is made (#1982) this could be reconsidered.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 02.01.2016 um 10:34 schrieb Christoph Hormann <notifications@github.com>:
> 
> The problem here is more that mapping tends to put the waterline at the outer edge of the reed bed and the actual water line within the reed is not mapped. This is perfectly understandable and also makes sense from a certain perspective but strictly seen there is no convention in OSM mandating that.


I d suggest to put the waterline at the actual position, so to comprise the reed bed, which not only solves the rendering issue but also is more consistent and closer to reality.

**Comment by mboeringa** [comment]
> I d suggest to put the waterline at the actual position, so to comprise the reed bed, which not only solves the rendering issue but also is more consistent and closer to reality.

Knowing what I know from the pretty common reedbeds here in the Netherlands, this will be nearly impossible. Reed can grow extremely dense both on land and in water (logged) positions. Including the reed bed as part of a water feature will therefore be as many times wrong as it may be right.

In fact, reed used to cover roofs is often collected from fields that may be water logged part of the year, with a soft peaty soil, and may be dry in other parts of the year. Other times, reed is collected in harsh winters, when the soil and water is frozen so its accessible.

**Comment by geaquinto** [comment]
I've reported a similar issue in #2025.

My suggestion is to keep rendering `natural=wetland` as a transparent layer with cyan dashes, but render every heavily used `wetland=*` with an icon and colour combination representative to each type of wetland. 

For instance, while `wetland=swamp` and `wetland=marsh` have this pattern (unique icons and colours identical to `natural=wood` and `natural=grassland`, respectively), `wetland=mangrove` and `wetland=reedbed` have only the unique icon. In my opinion, reedbeds should be coloured like `natural=scrub` or an already existing colour for `natural=*` with medium density.

**Comment by matkoniecz** [comment]
I think that primary problem is

> The problem here is more that mapping tends to put the waterline at the outer edge of the reed bed and the actual water line within the reed is not mapped.

Overriding water data and always rendering reed bed as on land or on water will worsen display in places with correct data. I think that it would not be desirable.

**Comment by imagico** [comment]
Thinking about it i would probably support adding a plain color background to all vegetated wetland types (grass for saltmash/reedbed, wood for swamp/mangrove) after #2066 and moving the ocean layers above the landcover base layer.

**Comment by Pikse** [comment]
> Overriding water data and always rendering reed bed as on land or on water will worsen display in places with correct data. I think that it would not be desirable.

I agree, it should be possible to distinguish coastline/waterline inside the redbeed, and transparency allows that. There are different ways to determine coastline. For instance, it may match the zero level of (local) datum. Imported coastline data by national land survey of Estonia at least partly uses this method. Generally, reedbed lies on both sides of this coastline.

**Comment by kocio-pl** [comment]
Nobody is currently involved in this problem, it's stale for almost 2 years. Would you like to propose a code solving it?

**Comment by Pikse** [comment]
> Nobody is currently involved in this problem, it's stale for almost 2 years. Would you like to propose a code solving it?

My comment was about why it'd be good to keep given style transparent. So if this question is addressed to me then I rather wouldn't propose a code "solving" it.

I found this issue as I was planning to map a large reed area. Now I doubt if I should as actual coastline might be lost because of that later on.

**Comment by Adamant36** [comment]
@kocio-pl, do you want to close this issue since it seems mostly resolved?

**Comment by kocio-pl** [comment]
Well, I don't understand wetlands too much, so it's hard for me to say what state is this problem in.

**Comment by jeisenbe** [comment]
wetland=mud is rendered with a semi-transparent brown, so that the coastline or edge of the natural=water is still visible. 

This has the disadvantage of creating 2 new colors, and could also lead to mixing with other backgrounds, as seen with military areas and mud currently.

Does anyone have thoughts about this?

**Comment by dieterdreist** [comment]
sent from a phone

> On 30. Dec 2018, at 00:49, jeisenbe <notifications@github.com> wrote:
> 
> wetland=mud is rendered with a semi-transparent brown, so that the coastline or edge of the natural=water is still visible.
> 
> This has the disadvantage of creating 2 new colors, and could also lead to mixing with other backgrounds, as seen with military areas and mud currently.
> 
> Does anyone have thoughts about this?
> 


I would not render “mud” at all, it isn’t on the wetland wikipage, IMHO for a reason: “mud” is odd amongst all those wetland types, it isn’t a type of wetland.

**Comment by Adamant36** [comment]
I always thought mud was a wierd tag. Its not really a thing in itself, but the temporary state of something else (dirt). How would someone confirm a muddy area is constantly that way anyway? Unless its the shore of a water body or part of a wetland. In which case it should just be tagged as those things or a shoreline.

**Comment by jeisenbe** [comment]
Sorry, my mistake. We render natural=mud and wetland=tidal_flat in
this way, but in landcover it's listed as wetland_mud. I think of the
rendering as "mud", as that is the name of the color in landcover.mss

Natural=mud and wetland=tidal_flat are both rendered with the same
sort of semi-transparent background fill color:
`rgba(203,177,154,0.3); // produces #e6dcd1 over @land`
- this has alpha 0.3, so it is 70% mixed with the background color
(usually land-color or water-color)

**Comment by jeisenbe** [comment]
@imagico, do you think mangroves and reedbeds could be rendered reasonably
well, by using a similar method?

I recall that your earlier recommendation was to switch to water polygons
first, and the render the ocean water above the landcover background
(without transparency).

**Comment by imagico** [comment]
As said before my approach would be to move to water polygons, do the layer re-ordering and start from there.  But this style has meanwhile moved into a different direction making this rather difficult.

**Comment by kocio-pl** [comment]
What is blocking move to water polygons? As far as i'm aware the problem was that somebody had problems with rendering on some setup, but I might confuse something, since I was not directly interested in it.

**Comment by kocio-pl** [comment]
Do you think that it's incompatible as an implementation or not doable with water polygons at all (as an idea)?

**Comment by imagico** [comment]
Regarding #3065 i don't know.  In general it seems compatibility with water polygons has not been a consideration for changes during the last 1-2 years so i have no idea what other problems there might be.  It could also depend on what kind of layer ordering you ultimately want to have.

**Comment by kocio-pl** [comment]
Unfortunately I'm not interested in wetlands etc, so can't help much, but it looks like it would need to check what might be conflicting here first. Probably not an easy task.

**Comment by kocio-pl** [comment]
I have just found again the issue that is blocking water tiles (#1982) in the first place: #2101.

**Comment by jeisenbe** [comment]
Now the ocean and water areas are rendered above landcover fill, so we can add a color to reedbeds (and mangroves) without problems.

What color should we use? Options would be:

A) Use the same light green used for grass and marshs: `@grass:` `#cdebb0`

- Advantages:  
1) Will match well with other green colors, does not require inventing or testing a new color
2) Reeds are somewhat similar to grasses

- Disadvantages
1) Makes it more difficult to distinguish marshes from reedbeds on land; only the pattern would be different
2) Reeds are taller and hardier than most other types of grass. It's usually somewhat easier to walk though a marsh than through a reedbed (although both are difficult)

B) Use brownish-green heath color `@heath:` `#d6d99f`

- Advantages:
1) Will match well with other green colors, does not require inventing or testing a new color
2) Reeds are not much taller than heath

- Disadvantages:
1) This color is already used for `wetland=bog`, and bogs are not very similar to reedbeds.
2) Reeds are slightly taller than the dwarf shrubs found in healthlands, and are related to grasses, not to shrubs which are woody plants.

C) B) Use mid-green-gray color of  `@scrub` `#c8d7ab`

Advantages:
1) Will match well with other green colors, does not require inventing or testing a new color
2) Reeds are sometimes nearly as tall as the shrubs found in scrub

Disadvantages
1) I would rather use `@scrub` `#c8d7ab` for mangroves, which are actually shrubs or trees
2) Reeds are slightly shorter than the shrubs found in scrublands, and are related to grasses, not to shrubs which are woody plants.

D) Use the light yellow-green color `@campsite`: `#def6c0`

Advantages:
1) Will match well with other green colors, does not require inventing or testing a new color
2) Similar to `@grass` color used for marshes, but slightly different

Disadvantages:
1) There is no direct similarity between campsites / caravan sites and reedbeds
2) I was thinking of using this color for `leisure=golf_course`

E) Find a new color, perhaps a shade of mid-green, greenish-brown, yellow-green or blue-green?

Advantages:
1) Would be more distinct from other types of wetland

Disadvantages
1) It's hard to find room for a new color in the greenish color space which isn't too close to another color. Remember when we changed scrub or allotments? That was really difficult

Right now my favorite idea is to use the light yellow-green color of `@campsite`: `#def6c0` for reedbeds

**Comment by jeisenbe** [comment]
# Test images:
0. current rendering
1. scrub - `#c8d7ab`
2. golf - `#b5e3b5`
3. allotments - `#c9e1bf`
4. campsite - `#def6c0`

## Current rendering (Italy)

wetland=reedbed around a small reservoir 
https://www.opensteetmap.org/#map=17/40.71722/16.15983
z17 current
![z17-tricarico-reedbed-before](https://user-images.githubusercontent.com/42757252/59075678-eb314b80-890c-11e9-9e88-66d6e8600e53.png)

Bradano river - reedbed where it enters a lake.
https://www.opensteetmap.org/#map=13/40.6261/16.4581

z13 before
![z13-bradano-reedbed-before](https://user-images.githubusercontent.com/42757252/59075676-e9678800-890c-11e9-8681-0cc53de906eb.png)

z15 before
![z15-bradano-reedbed-before](https://user-images.githubusercontent.com/42757252/59075677-ea001e80-890c-11e9-993f-4ecb59183307.png)

## 1) Reedbed with scrub color fill - `#c8d7ab`
z17 scrub color
![z17-tricarico-reedbed-scrubfill-after](https://user-images.githubusercontent.com/42757252/59075810-904c2400-890d-11e9-99e8-9753b2746d49.png)

z13 scrub color
![z13-bradano-reedbed-scrub-fill-after](https://user-images.githubusercontent.com/42757252/59075813-93471480-890d-11e9-8048-a18948f7bed1.png)

z15 scrub color
![z15-bradano-reedbed-scrubfill-after](https://user-images.githubusercontent.com/42757252/59075817-96da9b80-890d-11e9-9fcd-96eafee4e19f.png)

## 2) Reedbed golf color fill - `#b5e3b5`
z17 golf
![z17-tricarico-reedbed-golf-b5e3b5](https://user-images.githubusercontent.com/42757252/59076001-a4445580-890e-11e9-850c-077d84feccdb.png)

z13 golf
![z13-bradano-reedbed-golf-b5e3b5-after](https://user-images.githubusercontent.com/42757252/59076003-a6a6af80-890e-11e9-8982-b9f914593f16.png)

z15 golf
![z15-bradano-reedbed-golf-b5e3b5-after](https://user-images.githubusercontent.com/42757252/59076005-a9090980-890e-11e9-841f-832c04049a71.png)

## Reedbed allotments fill - `#c9e1bf`
z17 allotments
![z17-tricarico-reedbed-allotments-c9e1bf-after](https://user-images.githubusercontent.com/42757252/59076019-bfaf6080-890e-11e9-8891-9d462950f137.png)

z13 allotments
![z13-bradano-reedbed-allotments-c9e1bf-after](https://user-images.githubusercontent.com/42757252/59076022-c211ba80-890e-11e9-80a9-9fe3d34781ab.png)

z15 allotments
![z15-bradano-reedbed-allotments-c9e1bf-after](https://user-images.githubusercontent.com/42757252/59076025-c50cab00-890e-11e9-8721-0b64d980b763.png)

## Reedbed campsite fill - `#def6c0`
- slightly lighter and less saturated than grass / marsh / fen / wet meadow color

z17 campsite
![z17-tricario-reedbed-#def6c0-after](https://user-images.githubusercontent.com/42757252/59076058-ec637800-890e-11e9-9889-e1eaf9da6b4a.png)

z13 campsite
![z13-bradano-reedbed-campsite-#def6c0-after](https://user-images.githubusercontent.com/42757252/59076060-eec5d200-890e-11e9-9bac-9119531aa834.png)

z15 campsite
![z15-bradano-reedbed-campsite-#def6c0-afer](https://user-images.githubusercontent.com/42757252/59076071-f5ece000-890e-11e9-8564-e837c2707ddf.png)

**Comment by jeisenbe** [comment]
Tests in Zeeland, the Netherlands
https://www.opensteetmap.org/#map=16/51.4527/3.9174

### 0) Current rendering - z16
![z16-reedbed-farm-before](https://user-images.githubusercontent.com/42757252/59105698-72afa680-896f-11e9-986a-a793b155d487.png)

### 1) Reedbed with scrub color fill - `#c8d7ab`
![z16-reedbed-farm-scrub-fill](https://user-images.githubusercontent.com/42757252/59105724-8e1ab180-896f-11e9-9893-1da040413b67.png)

### 2) Reedbed golf color fill - `#b5e3b5`
![z16-reedbed-farm-golf-fill](https://user-images.githubusercontent.com/42757252/59105738-94109280-896f-11e9-9d46-538bd979823b.png)

### 3) Reedbed allotments fill - `#c9e1bf`
![z16-reedbed-farm-allotments-fill](https://user-images.githubusercontent.com/42757252/59105769-a5f23580-896f-11e9-940f-98950d73383e.png)

### 4) Reedbed campsite fill - `#def6c0`
![z16-reedbed-farms-meadow-campsite-fill](https://user-images.githubusercontent.com/42757252/59105785-ad194380-896f-11e9-8141-9779aedef95c.png)

**Comment by turnsole80** [comment]
3 for me

**Comment by jeisenbe** [comment]
As @imagico [suggested](https://github.com/gravitystorm/openstreetmap-carto/issues/2025#issuecomment-500160666) in #2025 here are test renderings of reedbeds with `@grass` fill - `#cdebb0`:

### 5) Reedbed `@grass` fill `#cdebb0`

z18 reedbeds before
![18-reedbeds-vs-marshes-before](https://user-images.githubusercontent.com/42757252/59547455-54066c80-8f7a-11e9-93ef-da12f3f7eb9d.png)

z18 reedbeds grass color after
![z18-reedbeds-marshes-grass-color-after](https://user-images.githubusercontent.com/42757252/59547452-20c3dd80-8f7a-11e9-9543-f322fe5dbffa.png)

z16 reedbed farm before
![z16-reedbed-farm-before](https://user-images.githubusercontent.com/42757252/59547504-cecf8780-8f7a-11e9-960b-74c8178b3ba4.png)

z16-reedbeds farm grass color after
![z16-reedbed-farm-grass-color](https://user-images.githubusercontent.com/42757252/59547487-b7909a00-8f7a-11e9-9aac-14c1a4ac7cb8.png)

**Comment by jeisenbe** [comment]
Bradano river - reedbed where it enters a lake.
https://www.opensteetmap.org/#map=13/40.6261/16.4581

z13 before
![z13-bradano-reedbed-before](https://user-images.githubusercontent.com/42757252/59075676-e9678800-890c-11e9-8681-0cc53de906eb.png)

z13 after - grass fill
![z13-bradano-grass-fill-reedbed](https://user-images.githubusercontent.com/42757252/59547620-9335bd00-8f7c-11e9-8ec1-77824bcf3552.png)

**Comment by Pikse** [comment]
To be clear, what is the effect of proposed changes on redbeed areas that lie on both sides of the coastline (discussed above)? E.g. is coastline still going to be distinguishable here: https://www.openstreetmap.org/#map=17/58.74029/23.78967?

**Comment by jeisenbe** [comment]
Water areas and seas are rendered above the landcover fill color, so the
background will still be blue for areas of reedbeds over water or outside
of the coastline. The coastline will be shown by the line between green and
blue.

**Comment by BertMule** [comment]
> Reed can grow extremely dense both on land and in water (logged) positions. Including the reed bed as part of a water feature will therefore be as many times wrong as it may be right.

Currently, if one considers a reed-bed to be **in** the water, then choosing where to draw the waterline is an unsatisfying choice, because of the rendering.

Either one draws the waterline:
- On the **waterside**, because that makes the bed better visible. Though that is not considered to be the shore.
- On the **landside** because that is considered the correct shore, but makes the bed 'transparent' and poorly visible.

Here is my current example of the latter.
https://www.openstreetmap.org/#map=18/53.19064/7.02622 

I propose to **always** give the bed a more solid colour, no matter on which side the shore is. 
Like the grass colour in examples above.

**Comment by imagico** [comment]
This does not look like a comment on the (closed) issue but a new request.  As such it seems, however, a duplicate of #3854.
